### PR TITLE
Require sklearn only for FastICA

### LINKED ISSUE #6062: Do we need scikit-learn for Infomax ICA?
Currently, we [require `scikit-learn` for all ICA algorithms](https://github.com/mne-tools/mne-python/blob/cffcf4b1aeaf04655f6917718029c59ff412c917/mne/preprocessing/ica.py#L316). However, I don't think that Infomax requires `scikit-learn`, only FastICA uses the implementation in this package. Not sure how we handle Picard, but I think it requires `scikit-learn` either.

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #6062.

### CODE REVIEW

**Comment by agramfort** [comment]
no test to update? like a test using infomax and that says that it requires sklearn?

**Comment by larsoner** [comment]
I would remove the `requires_sklearn` decorator in one of the parametrized functions and add `if method == 'fastica': try import sklearn: except ImportError: raise SkipTest(...)`.

**Comment by cbrnr** [comment]
`flake8` complains that `mne/preprocessing/tests/test_ica.py:130:13: F401 'sklearn' imported but unused`. How should I handle this?

**Comment by agramfort** [comment]
add a  `#noqa`

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -313,9 +313,9 @@ def __init__(self, n_components=None, max_pca_components=None,
                  "in 0.19. If you want to use Extended Infomax, specify "
                  "method='infomax' together with "
                  "fit_params=dict(extended=True).", DeprecationWarning)
-        if not check_version('sklearn', '0.15'):
-            raise RuntimeError('the scikit-learn package (version >= 0.15) '
-                               'is required for ICA')
+        if method == 'infomax' and not check_version('sklearn', '0.15'):
```
Comment (larsoner) [inline mne/preprocessing/ica.py]: shouldn't this be `if method == 'fastica' and ...`?

Comment (cbrnr) [reply mne/preprocessing/ica.py]: Of course.

**Comment by larsoner** [comment]
Thanks @cbrnr

**Pull request merged**